packages/java-edition/src/index.ts

```diff
@@ -156,6 +157,44 @@ export const initialize: core.ProjectInitializer = async (ctx) => {
 			},
 		},
 	)
+	const packFormats = new Map<number, McmetaVersion>()
+	for (const version of versions) {
+		if (version.type === 'release' && !packFormats.has(version.data_pack_version)) {
+			packFormats.set(version.data_pack_version, version)
+		}
+	}
+	mcdoc.runtime.registerAttribute(meta, 'pack_format', () => undefined, {
+		checker: (_, typeDef) => {
+			if (typeDef.kind !== 'literal' || typeof typeDef.value.value !== 'number') {
+				return undefined
+			}
+			const target = typeDef.value.value
+			return (node, ctx) => {
+				const targetVersion = packFormats.get(target)
+				if (!targetVersion) {
+					ctx.err.report(
+						localize('java-edition.pack-format.unsupported', target),
+						node,
+						core.ErrorSeverity.Warning,
+					)
+				} else if (targetVersion.id !== release) {
+					ctx.err.report(
+						localize('java-edition.pack-format.not-loaded', target, release),
+						node,
+						core.ErrorSeverity.Warning,
+					)
+				}
```

Comment:
instead of raising a warning in a file most people likely won't check, why don't we just forcibly delete the cache here? (or restart spyglass, whatever is needed to re-parse the `pack.mcmeta` and datapack contents)

Reply:
A user could have multiple pack.mcmeta files, or they could have an explicit `gameVersion` config set, or they may want to change the pack_format to match the currently loaded version instead.

Automatically restarting vscode or loading the correct version is out of scope for this PR, and will be handled when fixing #1071 and #1212